#164598708 postgresql database setup

caleb-42 opened the pull request:
#### What does this PR do?
- Create database connection to app
#### Description of Task to be completed?
- download postgresql node module
- clear all data structure dependent code
#### Any background context you want to provide?
- formerly written test will fail due empty implementation methods.
#### What are the relevant pivotal tracker stories?
- PT Story link - https://www.pivotaltracker.com/story/show/164598708

Commit:
chore(restructure): restructure folders for api/v2

- create api folder to accomodate version 1 and
version 2 of the api

[chore #164598708]

Commit:
chore: rebasing unto develop

[chore #164598708]

Commit:
chore(dbsetup): setup database coonnection

- Create SQL script for database creation.
- Create postgreSQL connection in db handler class.

[chore #164598708]

IyiKuyoro commented on src/database/db.sql:
```diff
@@ -0,0 +1,22 @@
+  password TEXT
+);
+CREATE TYPE _status as enum('read', 'unread', 'draft');
+  CREATE TABLE IF NOT EXISTS messages (
```
Fix this indentation issue.

IyiKuyoro commented on src/database/dbHandler.js:
```diff
@@ -2,237 +2,85 @@ import _ from 'lodash';
+    /* find any record in database */
   }
 
   generateJWT(user) {
```
Please remove this from this class. It will be more appropriate in a helper.

IyiKuyoro commented on src/database/dbHandler.js:
```diff
@@ -2,237 +2,85 @@ import _ from 'lodash';
+    /* create user using in user table */
   }
 
   async validateUser(guest, user) {
```
This should be in the sign-in controller.

IyiKuyoro reviewed (changes requested):
Just a few changes and this can be merged.

Commit:
chore(review): implement all review

- Fix db.sq file indentation issue.
- Remove generateJWT from dbhandler class into utilities file.
- Place user validation within User Sign in.

[chore #164598708]
